### ISSUE TITLE
Admin Toolbar Menus Inaccessible Implementation

### ISSUE THREAD

**Issue by avaaec** [pressbooks/pressbooks#1771]
### Steps to Reproduce

1. Go to admin interface for a book
2. Navigate to the Toolbar navigation region
2. Use the mouse to hover over this menu and observe the options
3. Using the keyboard navigate to the first Menu "My Catalog"
4. Try using space key to open the menu
5. Try using down arrow to open this menu
6. Use enter to open the menu
7. Try using the arrow keys to navigate within the menu
8. Use tab to navigate through the menu items
9. Try using space and right arrow to open sub menus
9. Use tab to move past the last menu item and notice that the menu stays open

**Expected behavior:** 
The menus should conform to the [WAI ARIA specification for menus / menu bars](https://www.w3.org/TR/wai-aria-practices-1.1/#menu).
This is slightly complicated by the fact that the menu buttons that open the menus are themselves links. This could be solved by redesigning the menus to be accessible.
Ultimately I think the admin Toolbar and Main Menu should be implemented as menubars, an accessible example of which can be found [here](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html). 

**Actual behavior:**
Although the menus "My Catalog" and "[Name of book]" are technically keyboard accessible using tab and enter, they do not follow the expectations of keyboard users to an extreme degree. The only way to open the menus (and submenus) is using the enter key. This breaks expectations because the element that opens the menu/submenu is itself a link. The [WAI ARIA authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/) clearly define that links should be opened with enter and that buttons (including menu buttons) should be activatable with space and enter. Menus/submenus should be openable with space, enter, and (recommended but optional) down and up arrow.
So as a keyboard user when I see a "menu button link" I already have broken expectations because the keyboard interaction for a link conflicts with the keyboard interaction for a menu button. 
Additionally a non-visual user would have extreme difficulty navigating this menu because it lacks proper aria menu labeling (see w3 example linked above) and because they would have no way knowing when they had navigated past the end of the table.

**Comment by SteelWagstaff** [comment]
Thanks for reporting, @avaaec. I believe this currently applies only to the menu item 'my catalogue', which should be fixed. I'd welcome further clarification if I'm mistaken.

**Comment by avaaec** [comment]
So sorry, when I originally filed this issue I actually missed some behavior when testing. Both menus are ```****technically****``` accessible but break expectations to a degree that I missed some of their behavior. I have updated the original ticket and title to reflect a more thorough testing.

**Comment by laurenray** [comment]
This is UW priority 0 issue.

**Comment by SteelWagstaff** [comment]
@laurenray Sorry if this is dumb question, but when you say this is priority 0 does that mean very high priority or very low priority?

**Comment by laurenray** [comment]
Hi @SteelWagstaff Priority 0 means very high priority. The prioritization is explained in the Evaluation section of the report we sent.  Many thanks.

**Comment by SteelWagstaff** [comment]
Great. Found the report. Thanks.

**Comment by SteelWagstaff** [comment]
@avaaec thanks for discussing with me in our meeting. More coming soon ...

**Comment by SteelWagstaff** [comment]
In the `#wpadminbar` that appears at the top of the page for logged in admin users, the .ab-top-menu displays up to five unique list items (shown in screenshot below).  
![Screenshot from 2019-12-07 14-48-02](https://user-images.githubusercontent.com/13485451/70380396-96dd4780-1900-11ea-8eb7-e6a0b5906bf2.png)
Each of these list items acts a both a link and a parent element to a pop-up menu which contains a number of list items. In each case **except** for 'My Catalog' the parent element links to the same place as the first item in the popup menu. To make this menu more accessible to users, perhaps we could 
1. add a 'My Catalog' link to the list that appears under the 'My Catalog' menu heading
2. remove the links from each of the top-level menu items and add the `aria-haspopup="true"` label directly to the parent `li`

**Comment by connermwallace** [comment]
University of Washington Accessibility Technology Services has inspected this issue and we have found that in addition to the fixes mentioned above, the submenu items are unable to be expanded using keyboard. This remains a UW ATS priority 0 issue.

**Comment by SteelWagstaff** [comment]
Proposed fix: https://github.com/pressbooks/pressbooks/issues/1893#issuecomment-592004606

**Comment by mcgratay** [comment]
Nav bar is now inconsistent. All other top-level nav items (Network Admin, rootsite title, and book title) all link to the first page in the dropdown when clicked. My Books does not link anywhere. I recommend either removing the links from all top-level nav items, or having "My Books" link to the Catalog.

**Comment by mcgratay** [comment]
💯
